Update event subscriber (#8483)
kyma-project/kyma: tools/event-subscriber/README.md

# E2E Subscriber

This is a sample subscriber app that simulates an event subscriber via exposing a set of APIs which can be used to receive and list events. It is meant to be only used in tests.

## Endpoints

| Endpoint                                     | Method | Usage                                                                                             |
|-------------------------------------------- |------ |------------------------------------------------------------------------------------------------- |
| `/ce`                                      | POST   | stores the received cloudevent                                                                    |
| `/ce`                                      | GET    | return list of all received cloudevents                                                           |

> | `/ce/<uuid>`                               | GET    | return cloud event with matching uuid. Returns 200 on success, 204 if no event was found          |

Author: and how would you differentiate then from endpoint not found?

> | `/ce/<uuid>`                               | GET    | return cloud event with matching uuid. Returns 200 on success, 204 if no event was found          |

Reviewer: As discussed: 404 or 204. Both works.

> | `/ce/<source>/<type>/<event-type-version>` | GET    | return all events matching source/type/version. Returns 200 on success, 204 if no event was found |

Reviewer 2: ```suggestion
| `/ce/<source>/<type>/<event-type-version>` | GET    | Returns all CloudEvents matching the source/type/version. Returns `200` on success, `204` if no event was found. |
```

> | `/`                                        | POST   | increase counter of received requests. Cloudevents sent to /ce do not increase this counter       |

Reviewer 2: ```suggestion
| `/`                                        | POST   | Increases the counter of received requests. CloudEvents sent to `/ce` do not increase this counter.     |
```

> | `/`                                        | GET    | return current counter value                                                                      |

Reviewer 2: ```suggestion
| `/`                                        | GET    | Returns the current counter value.                                                                     |
```

> | `/`                                        | DELETE | reset counter and list of received cloudevents                                                    |

Reviewer 2: ```suggestion
| `/`                                        | DELETE | Resets the counter and clears the list of received CloudEvents.                                                    |
```

## Commandline

| Flag                   | Usage          |
|---------------------- |-------------- |

> | `--port <int>` | listen on port |

Reviewer 2: ```suggestion
| `--port <int>` | Listens on o a provided port.|
```
